File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/parking/GetParkingsForFleetUseCase.java
package com.lattis.ellipse.domain.interactor.parking;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.Parking;
import com.lattis.ellipse.domain.repository.ParkingRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;


public class GetParkingsForFleetUseCase extends UseCase<List<Parking>> {

    private ParkingRepository parkingRepository;

    private int fleetId;

    @Inject
    public GetParkingsForFleetUseCase(ThreadExecutor threadExecutor,
                                      PostExecutionThread postExecutionThread,
                                      ParkingRepository parkingRepository) {
        super(threadExecutor, postExecutionThread);
        this.parkingRepository = parkingRepository;
    }

    public GetParkingsForFleetUseCase withFleetId(int fleetId) {
        this.fleetId = fleetId;
        return this;
    }

    @Override
    protected Observable<List<Parking>> buildUseCaseObservable() {
        return parkingRepository.getParkingSpotsForFleet(fleetId);
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/parking/GetParkingsForFleetUseCase.kt
package com.lattis.domain.usecase.parking

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.ParkingRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.Parking
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetParkingsForFleetUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val parkingRepository: ParkingRepository
) : UseCase<List<Parking>>(threadExecutor, postExecutionThread) {
    private var fleetId = 0
    fun withFleetId(fleetId: Int): GetParkingsForFleetUseCase {
        this.fleetId = fleetId
        return this
    }

    override fun buildUseCaseObservable(): Observable<List<Parking>> {
        return parkingRepository.getParkingSpotsForFleet(fleetId)
    }

}